src/components/button.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {MouseEvent, Component} from "react";
+import "../stylesheets/buttons.scss"
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {IconProp} from "@fortawesome/fontawesome-svg-core";
+import {isUndefined} from "lodash";
+
+/**
+ * Button props
+ * text: text to display on button
+ * icon: font awesome icon name [optional]
+ * onClick: onClick handler function
+ */
+export interface ButtonProps {
+    text: string;
+    icon: IconProp;
+    onClick(e: MouseEvent<HTMLElement>): void;
```

Comment:
Just a question, what determines the argument for onClick? I have string for some cases and don't know if it's appropriate. Should we always use MouseEvent?

Reply:
That onClick arg is based on onClick attribute on the <button> tag. Just Ctrl+click on "onClick" in `<button className="button" onClick={this.props.onClick} disabled={this.state.disabled}>` you'll see it needs a MouseEvent handler.